### ISSUE TITLE
How to detect Lapis environment in init_by_lua*?

### ISSUE THREAD

**Issue by vadi2** [leafo/lapis#423]
I need to read the Lapis configuration file in init_by_lua*, but the wrong configuration is being read since lapis hasn't loaded yet. Does the command-line argument get saved anywhere that I can retrieve it? LAPIS_ENVIRONMENT isn't defined yet at that point.

Thanks!

**Comment by leafo** [comment]
The configuration loading is done on demand. If you run `require("lapis.config").get()` it should load th e correct configuration for the environment in the same way that lapis does when it first loads.

I take it you're setting the configuration via the environment variable?

**Comment by vadi2** [comment]
It wasn't loading the right configuration for me when I was using that in the init_by_lua block - also LAPIS_ENVIRONMENT was still nil at that point. However nginx.conf.compiled was written by that point, so this worked:

```
io.input("nginx.conf.compiled")
local environment = io.read("*all"):match("env LAPIS_ENVIRONMENT=(%w+)")
local config  = require("lapis.config").get(environment)
io.input():close()
```

Would this be OK or is there anything better?

**Comment by leafo** [comment]
hmm, that will work but it's a pretty ugly workaround. It's possible that openresty doesn't expose environment variables in `init_by_lua`. Most likely because `init_by_lua` doesn't happen in a worker. You can try using `init_worker_by_lua` instead, but this will be called once for every worker. I can also set the environment variable in the shell so `init_by_lua` has something to read.

**Comment by vadi2** [comment]
Yeah I'd be happy to see the environment variable set in the shell - to be honest that's how I thought it was done already.
